.github/workflows/build_and_test.yml

```diff
@@ -91,3 +89,106 @@ jobs:
           path: |
             ~/${{ env.RESULT_PATH }}/results.html
             ~/${{ env.RESULT_PATH }}/results.xml
+
+  integration_test:
+    name: integration_test
+    # Running on Ubuntu 18.04 as Collectd python plugin required version GLIBC_2.29 which is not available on ubuntu-latest (20.04)
+    runs-on: ubuntu-18.04
+    needs: [build]
+    strategy:
+      matrix:
+        group: [1, 2, 3, 4]
+      fail-fast: false
+    steps:
+      - name: Check out the codebase.
+        uses: actions/checkout@v2
+
+      - name: Downloading image
+        uses: actions/download-artifact@v2
+        with:
+          name: image
+          path: ./dist
+
+      - name: Load agent image
+        run: docker load -i ./dist/image.tar
+
+      - name: Setup python
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ env.PYTHON_VERSION }}
+
+      - name: Caching dependency
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cache/pip
+          key: v1-pytest-${{ env.PYTHON_VERSION }}-${{ hashFiles(env.REQUIREMENTS_PATH) }}
+
+      - name: Install pytest
+        run: pip install -r "${{ env.REQUIREMENTS_PATH }}"
+
+      - uses: dorny/paths-filter@v2
+        id: filter
+        with:
+          filters: |
+            openstack:
+              - '**/openstack/**'
+              - '**/make-devstack-image'
+              - '**/run-devstack-image'
+              - '**/devstack/**'
+            conviva:
+              - '**/conviva/**'
+            jenkins:
+              - '**/jenkins/**'
+
+      - name: Setup integration test
+        run: |
+          export BUNDLE_DIR="$(pwd)/bundle"
+          export TEST_SERVICES_DIR="$(pwd)/test-services"
+          export AGENT_BIN="$BUNDLE_DIR/bin/signalfx-agent"
+          
+          mkdir -p "$BUNDLE_DIR"
+          cid=$(docker create quay.io/signalfx/signalfx-agent-dev:latest true)
+          docker export $cid | tar -C "$BUNDLE_DIR" -xf -
+          $BUNDLE_DIR/bin/patch-interpreter $BUNDLE_DIR
+          docker rm -fv $cid
+          [ -f "$AGENT_BIN" ] || (echo "$AGENT_BIN not found!" && exit 1)
+
+          MARKERS="integration";
+          if [ ${GITHUB_EVENT_NAME} != 'push' ] || [ ${GITHUB_REF##*/} != 'main' ]; then
```

Comment:
Were you able to confirm if these tests pass in github?  If not, please temporarily disable this section so that we can confirm that all tests pass before we merge.

Reply:
@jcheng-splunk yes I have tested this and it is passing test cases.
For testing I am triggering workflow for test branch by considering test branch as main. So I have made changes in workflow file accordingly  (`[ ${GITHUB_REF##*/} != 'main' ]` -> `[ ${GITHUB_REF##*/} != 'test' ]`)

Test Cases:
1. Push on _main_ branch -> all integration test should run : [Passed](https://github.com/nitaliya/signalfx-agent/runs/4786881473?check_suite_focus=true#step:9:22)
2. Pull request on _main_ or any branch -> specific integration test should run : [Passed](https://github.com/nitaliya/signalfx-agent/runs/4786957937?check_suite_focus=true#step:9:22)